trustmanager/x509utils.go

```diff
@@ -507,18 +505,18 @@ func X509PublicKeyID(certPubKey data.PublicKey) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	var finalAlgorithm data.KeyAlgorithm
+	pubKeyBytes, err := x509.MarshalPKIXPublicKey(cert.PublicKey)
+	if err != nil {
+		return "", err
+	}
 
+	var key data.PublicKey
 	switch certPubKey.Algorithm() {
 	case data.ECDSAx509Key:
-		finalAlgorithm = data.ECDSAKey
+		key = data.NewECDSAPublicKey(pubKeyBytes)
```

Comment:
Yes, and this was true before. I don't think it causes any meaningful vulnerabilities but we should sit down and talk through the possibilities.

I think the only place it would be a problem would be if somebody was running a modified version of the code that set the wrong key type. As long as the initial keys are good (which they will be as long as the user isn't running a modified binary, in which case all safety has gone out the window anyway), I think key continuity should ensure an attacker doesn't have an angle related to setting the wrong key type (particularly as we closed any holes similar to that JWT problem).